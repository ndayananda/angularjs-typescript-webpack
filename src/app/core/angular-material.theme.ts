/**
 * Configuring of the default theme is done by using the $mdThemingProvider during application configuration.
 *
 * @param {angular.material.IThemingProvider} $mdThemingProvider
 *
 * @return void
 */

export const customThemeConfig = ($mdThemingProvider: angular.material.IThemingProvider): void => {
    'ngInject';
 
     $mdThemingProvider.definePalette('appGreyThemePalette', getGreyThemePalette());
     $mdThemingProvider.definePalette('appBlueThemePalette', getBlueThemePalette());
 
     $mdThemingProvider.theme('default')
         .primaryPalette('appBlueThemePalette')
         .accentPalette('appBlueThemePalette');
 };
 
 const commonPaletteConfig = {
     'contrastDefaultColor': 'light',    // whether, by default, text (contrast)
     // on this palette should be dark or light
 
     'contrastDarkColors': ['50', '100', //hues which contrast should be 'dark' by default
     '200', '300', '400'],
     'contrastLightColors': undefined    // could also specify this if default was 'dark'
 };
 
 const getGreyThemePalette = () => {
     var palette = {
         '50': 'f2f2f2',
         '100': 'e6e6e6',
         '200': 'cccccc',
         '300': 'b3b3b3',
         '400': '999999',
         '500': '808080',
         '600': '666666',
         '700': '4d4d4d',
         '800': '333333',
         '900': '1a1a1a',
         'A100': '9E9E9E',
         'A200': '757575',
         'A400': '616161',
         'A700': '424242'
     }
 
     palette = Object.assign({}, palette, commonPaletteConfig);
 
     return palette;
 };
 
 const getBlueThemePalette = () => {
     var palette = {
         '50': '76c3f6',
         '100': '5eb8f5',
         '200': '2ea3f2',
         '300': '0e8bdf',
         '400': '0b6daf',
         '500': '095e97',
         '600': '084f7f',
         '700': '064067',
         '800': '05314f',
         '900': '032237',
         'A100': '0b6daf',
         'A200': '066eb4',
         'A400': '026fb8',
         'A700': '2962FF',
     }
 
     palette = Object.assign({}, palette, commonPaletteConfig);
 
     return palette;
 };