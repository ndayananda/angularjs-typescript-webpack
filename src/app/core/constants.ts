/**
 * 	constants are read only pieces of data.
 * 	constants can be injected anywhere including configuration calls.
 *
 */

export const appConstants: Object = {
    appName: 'Angularjs Typescript Webpack single page application',
    version: '[AIV]{version}[/AIV]',
    defaultPaginationProperties: {
        start: '0',
        limit: '100',
    },
    apiUrl: {
        users: {
            GET_USERS: 'http://localhost:3000/users'
        }
    }
};