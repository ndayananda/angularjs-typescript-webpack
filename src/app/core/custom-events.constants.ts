/**
 * 	customEvents is a constant object
 * 	Declare all the custom events here
 *
 */

export const customEvents: Object = {
    collapsiblePanel: {
        PANEL_COLLAPSED: 'PANEL_COLLAPSED',
        PANEL_EXPANDED: 'PANEL_EXPANDED'
    }
};