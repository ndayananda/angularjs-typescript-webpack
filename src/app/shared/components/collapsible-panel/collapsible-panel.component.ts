import { IComponentOptions, Injectable, IControllerConstructor } from 'angular';

import { CollapsiblePanelController } from './collapsible-panel.controller';
import './collapsible-panel.scss';

const templateUrl = require('./collapsible-panel.template.html');
/**
 * This is a CollapsiblePanelComponent
 *
 * @export CollapsiblePanelComponent
 * @class CollapsiblePanelComponent
 * @implements {IComponentOptions}
 */
export class CollapsiblePanelComponent implements IComponentOptions {
    static NAME: string = 'collapsiblePanel';

    templateUrl: Injectable<(...args: any[]) => string>;
    controller: Injectable<IControllerConstructor>;
    bindings: {[boundProperty: string]: string};
    transclude: {[slot: string]: string};

    constructor() {
        this.templateUrl = templateUrl;
        this.transclude = {
            toolbarContent: '?cpToolbarContent',
            content: '?cpContent',
            title: '?cpTitle'
        };
        this.bindings = {
            vertical: '@',
            panelWidth: '@',
            className: '@',
            uuid: '@',
            collapsed: '<',
            align: '@'
        };
        this.controller = CollapsiblePanelController;
    }
}