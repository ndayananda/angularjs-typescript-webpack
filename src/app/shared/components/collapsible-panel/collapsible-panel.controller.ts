import * as angular from 'angular';
import { IComponentController, IScope, IRootElementService, ITimeoutService } from 'angular';

import { UuidService } from './../../service/uuid/uuid.service';

export class CollapsiblePanelController implements IComponentController {

    private vertical: string;
    private panelWidth: string;
    private className: string;
    private uuid: string;
    private align: string;
    private collapsed: boolean = true;

    constructor(
        private uuidService: UuidService,
        private $element: IRootElementService,
        private $timeout: ITimeoutService,
        private customEvents
    ) {
        'ngInject';
    }

    $onInit() {
        this.vertical = this.vertical || 'false';
        this.panelWidth = this.panelWidth || '260';
        this.className = this.className || '';
        this.uuid = this.uuid || this.uuidService.generate();
        this.align = this.align || 'right';
        this.collapsed = this.collapsed || false;

        if( !this.isVerticalPanel() )
            angular.element(() => {
                this.$timeout(() => this.togglePanelHeight(), 2000);
            })
    }

    private isVerticalPanel(): boolean {
        return (this.vertical === 'true') ? true : false;
    }

    private toggle($event) {
        this.collapsed = !this.collapsed;

        if( !this.isVerticalPanel() )
            this.togglePanelHeight();
    }

    private isCollapsed(): boolean {
        return this.collapsed;
    }

    private isAlginLeft(): boolean {
        return (this.align === 'left') ? true : false;
    }

    private getPanelWidth(): string {
        let width = '';
        if( this.isVerticalPanel() )
            width = this.collapsed ? '43px' : this.panelWidth + 'px';

        return width;
    }

    private togglePanelHeight() {
        let element = this.$element[0].querySelector('.cp-content');
        angular.element(element).css('height', this.collapsed ? 0 : `${ element.scrollHeight + 1 }px`);
    }
}