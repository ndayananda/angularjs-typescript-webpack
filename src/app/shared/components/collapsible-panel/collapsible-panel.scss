@import './../../sass/variables';

.cp {
    border: 1px solid $primary-border-color;
    transition: all ease 0.4s;

    &.no-border {
        border-width: 0;
    }

    &.cp-vertical-panel {
        width: 260px;
        background: lighten($primary-color, 30%);
        position: relative;
        height: calc(100% - 1px);

        .md-panel-outer-wrapper {
            &._md-panel-shown {
                .loader-panel {
                    position: absolute;
                }
            }
        }

        .cp-action-bar {
            line-height: 4px;
            padding:0 5px;
            border-bottom: 1px solid $primary-border-color;
            background: lighten($primary-color, 28%);
            color: lighten($primary-color, 90%);
            text-align: left;

            .fa-xs {
                font-size: .45rem;
            }
        }

        .cp-toolbar {
            width: 42px;
            background: lighten($primary-color, 30%);
            position: absolute;
            top: 11px;
            bottom: 0;
            transition: all ease 0.4s;
            overflow: hidden;

            li {
                padding: 10px 5px;
                border-bottom: 1px solid $primary-border-color;

                a {
                    color: $text-color;
                    display: block;
                    padding: 3px;
                    font-size: 12px;
                    border: 1px solid transparent;
                    text-align: center;

                    &:hover {
                        background: lighten($primary-color, 85%);
                        border-color: $primary-border-color;
                        border-radius: 3px;
                        text-decoration: none;
                    }

                    .fa {
                        font-size: 15px;
                        line-height: 15px;
                    }
                }
            }

            .version {
                background: $primary-border-color;
                text-align: center;
                font-size: 10px;
                font-weight: 500;
                position: absolute;
                bottom: 0;
                left: 0;
                right: 0;
            }
        }

        &.align-left {
            left: 0;
            border-width: 0 1px 1px 0;

            .cp-action-bar {
                text-align: right;
            }

            &.expand {
                .cp-action-bar {
                    .fa-chevron-left {
                        display: none;
                    }
                }
                .cp-content {
                    display: none;
                }
            }

            &.collapse {
                .cp-action-bar {
                    .fa-chevron-right {
                        display: none;
                    }
                }
                .cp-toolbar {
                    width: 0;
                }
            }

            .cp-toolbar {
                left: 0;
            }
        }

        &.align-right {
            right: 0;
            border-width: 0 0 1px 1px;

            &.expand {
                .cp-action-bar {
                    .fa-chevron-right {
                        display: none;
                    }
                }
                .cp-content {
                    display: none;
                }
            }

            &.collapse {
                .cp-action-bar {
                    .fa-chevron-left {
                        display: none;
                    }
                }
                .cp-toolbar {
                    width: 0;
                }
            }

            .cp-toolbar {
                right: 0;
            }
        }
    }

    &.cp-horizontal-panel {
        .toolbar {
            border-bottom: 0;
            
            .cp-toggle {
                position: absolute;
                right: 10px;
                top: 14px;
            }
        }

        .cp-content {
            overflow-y: auto;
            overflow-x: hidden;
            border-top: 1px solid $primary-border-color;
            transition: all ease 0.4s;
        }

        &.expand {
            .fa-chevron-up {
                display: none;
            }
            .cp-content {
                opacity: 0;
                height: 0;
                border: 0;
                overflow: hidden;
            }
        }

        &.collapse {
            .fa-chevron-down {
                display: none;
            }
            .cp-content {
                opacity: 1;
            }
        }
    }
}