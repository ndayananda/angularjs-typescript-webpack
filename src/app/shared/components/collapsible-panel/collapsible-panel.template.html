<div class="cp {{ $ctrl.className }}" ng-class="{
        'cp-vertical-panel': $ctrl.isVerticalPanel(),
        'cp-horizontal-panel': !$ctrl.isVerticalPanel(),
        'expand': $ctrl.isCollapsed(),
        'collapse': !$ctrl.isCollapsed(),
        'align-left': $ctrl.isAlginLeft(),
        'align-right': !$ctrl.isAlginLeft()
    }" ng-style="{'width': $ctrl.getPanelWidth()}">
    <div class="cp-action-bar" ng-click="$ctrl.toggle($event)" ng-if="$ctrl.isVerticalPanel()">
        <i class="fa fa-chevron-left" aria-hidden="true"></i><i class="fa fa-chevron-left" aria-hidden="true"></i>
        <i class="fa fa-chevron-right" aria-hidden="true"></i><i class="fa fa-chevron-right" aria-hidden="true"></i>
    </div>
    <div class="cp-toolbar" ng-if="$ctrl.isVerticalPanel()">
        <div class="version">V-[AIV]{version}[/AIV]</div>
        <div ng-transclude="toolbarContent"></div>
    </div>
    <div class="cp-title" ng-if="!$ctrl.isVerticalPanel()">
        <div class="toolbar">
            <div ng-transclude="title"></div>
            <a href="" class="cp-toggle" ng-click="$ctrl.toggle($event)">
                <i class="fa fa-chevron-down" aria-hidden="true"></i>
                <i class="fa fa-chevron-up" aria-hidden="true"></i>
            </a>
        </div>
    </div>
    <div class="cp-content">
        <div ng-transclude="content"></div>
    </div>
</div>