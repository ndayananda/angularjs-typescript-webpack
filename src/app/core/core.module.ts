import * as angular from 'angular';
import uiRouter from '@uirouter/angularjs';
import * as ngMaterial from 'angular-material';
import * as ngAnimate from 'angular-animate';
import * as ngMessages from 'angular-messages';
import * as ngSanitize from 'angular-sanitize';
import * as ngAria from 'angular-aria';
import { ngTableModule } from 'ng-table';
import 'ng-table/bundles/ng-table';

import { appConstants } from './constants';
import { customEvents } from './custom-events.constants';
import { routeConfig } from './route.config';
import { customThemeConfig } from './angular-material.theme';
import { DataService } from './../shared/service/data-service.service';
import { UuidService } from './../shared/service/uuid/uuid.service';
import { CollapsiblePanelComponent } from './../shared/components/collapsible-panel/collapsible-panel.component';

/*
 * Core module acts as a heart of the application
 * here we inject all the important dependencies
 * which can be used across the modules, like services, resusable components etc.,
 */

export const appCore = angular
    .module('app.core', [
        /*
        * Angular modules
        */
        uiRouter,
        ngAnimate,
        ngSanitize,
        ngAria,
        ngMaterial,
        ngMessages,
        ngTableModule.name,

        /*
        * Our reusable cross app code modules
        */
        
    ])
    .constant('appConstants', appConstants)
    .constant('customEvents', customEvents)
    .constant('appConfig', {})
    .config(routeConfig)
    .config(customThemeConfig)
    .config(($animateProvider) => {
        'ngInject';
        $animateProvider.classNameFilter(/angular-animate/);
    })
    .service(DataService.NAME, DataService)
    .service(UuidService.NAME, UuidService)
    .component(CollapsiblePanelComponent.NAME, new CollapsiblePanelComponent())
    .name;

