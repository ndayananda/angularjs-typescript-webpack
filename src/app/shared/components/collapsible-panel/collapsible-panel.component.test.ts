import * as angular from 'angular';
import * as ngMock from 'angular-mocks';
import { appCore } from './../../../core/core.module';
import { CollapsiblePanelComponent } from './collapsible-panel.component';


beforeEach ( angular.mock.module(appCore) );

let $ctrl, element, broadcastService, template, $rootScope, $compile, $scope;

beforeEach (
	angular.mock.inject( (_$rootScope_, _$compile_, _$componentController_, _broadcastService_) => {
        $rootScope = _$rootScope_;
        $compile = _$compile_;

        $scope = $rootScope.$new();
		broadcastService = _broadcastService_;
		$ctrl = _$componentController_("collapsiblePanel", {
			$scope,
			broadcastService
        });

        element = angular.element('<collapsible-panel></collapsible-panel>');
		element = $compile(element)($scope);
        $scope.$apply();
	})
)

describe("Test component", () => {
	it("view model should be defined", () => {
		expect($ctrl).toBeDefined()
	});

	it("should default align to left", () => {
		expect($ctrl.align).toBe("left")
	});

	it("should render default dom", () => {
		expect(element).toBeDefined();
		expect(element[0]).toMatchSnapshot();
	});
})