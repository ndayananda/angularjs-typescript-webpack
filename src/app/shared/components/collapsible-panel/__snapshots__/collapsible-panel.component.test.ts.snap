// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component should render default dom 1`] = `
<collapsible-panel
  class="ng-scope ng-isolate-scope"
>
  <div
    class="cp  cp-horizontal-panel collapse align-right"
    ng-class="{
        'cp-vertical-panel': $ctrl.isVerticalPanel(),
        'cp-horizontal-panel': !$ctrl.isVerticalPanel(),
        'expand': $ctrl.isCollapsed(),
        'collapse': !$ctrl.isCollapsed(),
        'align-left': $ctrl.isAlginLeft(),
        'align-right': !$ctrl.isAlginLeft()
    }"
    ng-style="{'width': $ctrl.getPanelWidth()}"
  >
    
    
    <!-- ngIf: $ctrl.isVerticalPanel() -->
    
    
    <!-- ngIf: $ctrl.isVerticalPanel() -->
    
    
    <!-- ngIf: !$ctrl.isVerticalPanel() -->
    <div
      class="cp-title ng-scope"
      ng-if="!$ctrl.isVerticalPanel()"
    >
      
        
      <div
        class="toolbar"
      >
        
            
        <div
          ng-transclude="title"
        />
        
            
        <span
          class="cp-toggle"
          ng-click="$ctrl.toggle($event)"
          role="button"
          tabindex="0"
        >
          
                
          <i
            aria-hidden="true"
            class="fa fa-chevron-down"
          />
          
                
          <i
            aria-hidden="true"
            class="fa fa-chevron-up"
          />
          
            
        </span>
        
        
      </div>
      
    
    </div>
    <!-- end ngIf: !$ctrl.isVerticalPanel() -->
    
    
    <div
      class="cp-content"
    >
      
        
      <div
        ng-transclude="content"
      />
      
    
    </div>
    

  </div>
</collapsible-panel>
`;
